source/extensions/common/wasm/context.cc

```diff
@@ -1691,8 +1691,17 @@ WasmResult Context::sendLocalResponse(uint32_t response_code, std::string_view b
       if (local_reply_sent_) {
         return;
       }
+      // C++, Rust and other SDKs use -1 (InvalidCode) as the default value if gRPC code is not set,
+      // which should be mapped to nullopt in Envoy to prevent it from sending a grpc-status trailer
+      // at all.
+      absl::optional<Grpc::Status::GrpcStatus> grpc_status_code =
+          Grpc::Status::WellKnownGrpcStatus(grpc_status) ==
```

Comment:
This is undefined behavior I think for invalid values, e.g. 20? Can we change this to:
```c++
abs::optional<...> grpc_status_code = absl::nullopt;
if (grpc_status >= Ok && grpc_status <= MaximumKnown) {
  grpc_status_code = Grpc::Status::WellKnownGrpcStatus(grpc_status);
}
```

Reply:
Make sense, thanks

Reply:
This is interesting and it was interesting due to this enum is unscoped enum and the underlying type must contains it values, it must be at lease signed char

 > std 7.2.7
 For an enumeration whose underlying type is not fixed, the underlying type is an integral type that can
represent all the enumerator values defined in the enumeration.

then 20 can be assigned to the enumeration. 

> std 7.2.8 
It is possible to define an enumeration that has values not defined by any of its
enumerators. If the enumerator-list is empty, the values of the enumeration are as if the enumeration had a
single enumerator with value 0.

but only if it fits in the underlying type.

anyway it was my bad to cast an uint32 into an enum without fixed underlying type, specially one that is unsigned into an enum that should be signed to hold the -1 value. 

I should have done what you are proposing in the first place, Thanks again, and it is always good to review the standard.